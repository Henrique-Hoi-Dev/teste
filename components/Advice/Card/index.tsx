import { Link } from "@/components/Common/Link";
import { StrapiImage } from "@/components/Common/StrapiImage";
import { Card as CardAdvice } from "@/endpoints/fetchAdvice";

import styles from "./styles.module.scss";

export function Card(props: CardAdvice) {
  const { title, subTitle, description, iconString, icon, link } = props;

  return (
    <li className={styles["card"]}>
      <div className={styles["card__container"]}>
        <div className={styles["card__container__img"]}>
          <StrapiImage
            src={iconString}
            alt={icon.data.attributes.name}
            loading="eager"
          />
        </div>
        <h1>{title}</h1>
        <h5>{subTitle}</h5>
        <p>{description}</p>
        <div className={styles["card__container__button"]}>
          {link && (
            <Link variant="contained" href={link.Url} target={link.Target}>
              {link.Label}
            </Link>
          )}
        </div>
      </div>
    </li>
  );
}
