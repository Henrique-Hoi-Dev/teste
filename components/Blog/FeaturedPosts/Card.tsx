import { Author, Categorys } from "@/components/Blog";
import Link from "next/link";
import { useRouter } from "next/router";
import styles from "./card.styles.module.scss";

interface Props {
  background: string;
  slug: string;
  title: string;
  author: string;
  categorys: Array<{
    id: number;
    category: string;
  }>;
  className?: string;
  target?: string;
  customRoute?: boolean;
}

export function Card(props: Props) {
  const {
    background,
    slug,
    title,
    author,
    categorys,
    className,
    target,
    customRoute = false,
  } = props;

  const router = useRouter();

  const renderContent = () => (
    <li
      style={{
        background: background,
      }}
      className={`${styles["featured-posts__post"]} ${className}`}
      role="button"
      onClick={customRoute ? () => {} : () => router.push(`/blog/${slug}`)}
    >
      <strong>
        <span> {title} </span>
      </strong>
      <div className={styles["featured-posts__info"]}>
        <Author author={author} />
        <Categorys categorys={categorys} />
      </div>
    </li>
  );

  return customRoute ? (
    <Link href={slug || ""}>{renderContent()}</Link>
  ) : (
    renderContent()
  );
}
