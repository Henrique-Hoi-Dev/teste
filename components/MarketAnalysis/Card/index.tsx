import { Link } from '@/components/Common/Link'
import { Card } from '@/endpoints/fetchAdvice'
import { createStrapiImageUrl } from '@/utils/createStrapiImageUrl'
import ReactMarkdown from 'react-markdown'

import styles from './styles.module.scss'

export function Card(props: Card) {
  const { title, subTitle, description, link, downloadReport } = props

  return (
    <li className={styles['card']}>
      <div className={styles['card__container']}>
        <div>
          <h1>{title}</h1>
          <h5>{subTitle}</h5>
          <ReactMarkdown>{description}</ReactMarkdown>
        </div>
        <div className={styles['card__container__button']}>
          {link && (
            <Link variant="contained" href={link.Url} target={link.Target}>
              {link.Label}
            </Link>
          )}
          {downloadReport?.media.data?.attributes?.url && (
            <Link
              variant="contained"
              href={createStrapiImageUrl(
                downloadReport?.media?.data?.attributes?.url,
              )}
              target={'_blank'}
            >
              {downloadReport.title}
            </Link>
          )}
        </div>
      </div>
    </li>
  )
}
