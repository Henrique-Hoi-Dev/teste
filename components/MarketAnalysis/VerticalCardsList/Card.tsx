/* eslint-disable react-hooks/exhaustive-deps */
import { Link } from "@/components/Common/Link";
import React, { useEffect, useRef, useState } from "react";
import { StrapiImage } from "@/components/Common/StrapiImage";
import { StrapiImage as StrapiImageType } from "@/interfaces/StrapiImage";
import { StrapiLink } from "@/interfaces/StrapiLink";
import ReactMarkdown from "react-markdown";

import styles from "./styles.module.scss";

export interface CardProps {
  title: string;
  subTitle?: string;
  description: string;
  link?: StrapiLink;
  index: number;
  length: number;
  image: StrapiImageType;
}

export function Card({
  title,
  description,
  link,
  index,
  image,
  subTitle
}: CardProps) {
  
  const ref = useRef<HTMLLIElement>(null);
  const [isVisible, setVisible] = useState(false);

  const observerRef = useRef<IntersectionObserver | null>(null);

  useEffect(() => {
    observerRef.current = new IntersectionObserver(([entry]) => {
      setVisible(entry.isIntersecting);
      setTimeout(() => {
        observerRef.current?.disconnect();
      }, 2000);
    });
  }, []);

  useEffect(() => {
    if (observerRef?.current && ref?.current) {
      observerRef.current.observe(ref.current);
    }

    return () => {
      if (observerRef?.current && ref?.current) {
        observerRef.current.disconnect();
      }
    };
  }, [ref]);

  return (
    <li
      ref={ref}
      key={title}
      id={`investment-${index}`}
      className={isVisible ? styles["card__animated"] : ""}
    >
      <div className={styles["column"]}>
        <h2>{title}</h2>
        {subTitle && <h4>{subTitle}</h4>}

        <ReactMarkdown>{description}</ReactMarkdown>
        {link && (
          <Link
            href={link.Url}
            className={styles["link-button"]}
            variant="contained"
          >
            {link.Label}
          </Link>
        )}
      </div>
      <StrapiImage src={image.data.attributes.url} alt={title} loading="lazy" />
    </li>
  );
}
