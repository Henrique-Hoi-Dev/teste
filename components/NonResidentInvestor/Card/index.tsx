import { StrapiImage } from "@/components/Common/StrapiImage";
import styles from "./styles.module.scss";

interface InfoCardProps {
  title: string;
  description?: string;
  icon?: string;
  alt?: string;
}

export function Card({
  title,
  description,
  icon,
  alt
}: InfoCardProps) {
  return (
    <div className={styles["card"]}>
      <div className={styles["card__icon-title"]}>
        <picture>
          <StrapiImage src={icon} alt={alt} />
        </picture>
        <div className={styles["card__title"]}>
          <h1>{title}</h1>
          <hr />
        </div>
      </div>
      <div className={styles["card__description"]}>
        <p>{description}</p>
      </div>
    </div>
  );
}
