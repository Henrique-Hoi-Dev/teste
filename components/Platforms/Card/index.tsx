import { Link } from "@/components/Common/Link";
import { ReactMarkdown } from "react-markdown/lib/react-markdown";
import styles from "./styles.module.scss";

interface CardProps {
  title: string;
  description: string;
  labelButton: string;
  href: string;
}

export function Card(props: CardProps) {
  const {
    title,
    description,
    labelButton,
    href,
  } = props;

  return (
    <li className={styles.card}>
      <h3 className={styles["card__title"]}>{title}</h3>

      <ReactMarkdown className={styles["card__description"]}>{description}</ReactMarkdown>

      <Link href={href} variant="contained">
        {labelButton}
      </Link>
    </li>
  );
}
