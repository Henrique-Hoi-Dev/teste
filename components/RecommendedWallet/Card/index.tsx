import { Link } from "@/components/Common/Link";
import { StrapiImage } from "@/components/Common/StrapiImage";
import { Card } from "@/endpoints/fetchRecommendedWalletPageData";

import styles from "./styles.module.scss";

export function CardCarousel({
  title,
  description,
  link,
  avatar1,
  avatar2,
  date,
  titleAvatar,
  icon,
  graphicIcon1,
  graphicIcon2,
  graphicNumber1,
  graphicNumber2,
  graphicText1,
  graphicText2
}: Card) {
  return (
    <li className={styles["card-container"]}>
      <h1>{title}</h1>
      <h4>{date}</h4>
      <p>{description}</p>
      <div className={styles["card-container__bottom-container"]}>
        <div className={styles["card-container__graphic-outer-container"]}>
          <hr />
          <div className={styles["card-container__graphic-container"]}>
            <div
              className={styles["card-container__graphic-container__graphic1"]}
            >
              <h5>
                <StrapiImage src={graphicIcon1.data.attributes.url} />
                {graphicNumber1}
              </h5>
              <p>{graphicText1.toUpperCase()}</p>
            </div>
            <div
              className={styles["card-container__graphic-container__graphic2"]}
            >
              <h5>
                <StrapiImage src={graphicIcon2.data.attributes.url} />
                {graphicNumber2}
              </h5>
              <p>{graphicText2.toUpperCase()}</p>
            </div>
          </div>
          <hr />
        </div>
        <div className={styles["card-container__avatars"]}>
          <StrapiImage src={icon.data.attributes.url} />
          <StrapiImage src={avatar1.data.attributes.url} />
          <StrapiImage src={avatar2.data.attributes.url} />
          <h5>{titleAvatar}</h5>
        </div>
        <Link variant="contained" href={link.Url} target={link.Target}>
          {link.Label}
        </Link>
      </div>
    </li>
  );
}
