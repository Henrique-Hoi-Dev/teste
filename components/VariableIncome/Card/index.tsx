import { Link } from "@/components/Common/Link";
import { StrapiImage } from "@/components/Common/StrapiImage";
import { StrapiLink } from "@/interfaces/StrapiLink";

import styles from "./styles.module.scss";

interface Card {
  title: string;
  description: string;
  link: StrapiLink;
  icon: string;
  alt: string;
  noBorderRadius?: boolean;
}

export function Card(props: Card) {
  return (
    <li
      className={styles["card-container"]}
      style={{
        borderRadius: props.noBorderRadius ? 0 : "1rem"
      }}
    >
      <div className={styles["card-container__icon"]}>
        <StrapiImage src={props.icon} alt={props.alt} />
        <h4>{props.title}</h4>
      </div>
      <p>{props.description}</p>
      <div>
        {props.link && (
          <Link
            variant="contained"
            href={props.link.Url}
            target={props.link.Target}
          >
            {props.link.Label}
          </Link>
        )}
      </div>
    </li>
  );
}
