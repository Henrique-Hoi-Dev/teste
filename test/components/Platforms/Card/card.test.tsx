import { Card } from "@/components/Platforms/Card";
import { render, screen } from "@testing-library/react";

const cardPropsMock = {
  img: "cedro.png",
  alt: "cedro",
  title: "Start trade",
  price: "R$ 39,90/mês",
  description: `Versão Start da Fast trade na medida para atender o trader que procura
   custo benefício aliados a boas funcionalidades gráficas, ferramentas
   como Times & Trades, Chart Trading, Resumo das Bolsas entre outros.`,
  demoPeriod: "*Demonstrativo: 7 dias",
  buyButtonText: "Contratar",
  href: "https://corretora.miraeasset.com.br/Home/Login",
};

describe("Platforms card", () => {
  test("Should match snapshot", () => {
    const { container } = render(<Card {...cardPropsMock} />);
    expect(container).toMatchSnapshot();
  });

  test("Should render all the required infos", () => {
    render(<Card {...cardPropsMock} />);

    const img = screen.getByRole("img");
    expect(img).toHaveProperty(
      "src",
      expect.stringContaining(cardPropsMock.img)
    );

    expect(screen.getByText(cardPropsMock.title)).toBeInTheDocument();
    expect(screen.getByText(cardPropsMock.price)).toBeInTheDocument();
    expect(
      screen.getByText(cardPropsMock.demoPeriod, { exact: false })
    ).toBeInTheDocument();
    expect(screen.getByText(cardPropsMock.demoPeriod)).toBeInTheDocument();

    const buyButton = screen.getByRole("link", {
      name: cardPropsMock.buyButtonText,
    });
    expect(buyButton).toHaveProperty("href", cardPropsMock.href);
  });
});
