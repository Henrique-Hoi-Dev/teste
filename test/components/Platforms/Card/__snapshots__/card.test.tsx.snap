// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Platforms card Should match snapshot 1`] = `
<div>
  <li
    class="card"
  >
    <div
      class="card__img-container"
    >
      <picture>
        <img
          alt="cedro"
          src="undefinedcedro.png"
        />
      </picture>
    </div>
    <h3
      class="card__title"
    >
      Start trade
    </h3>
    <span
      class="card__price"
    >
      R$ 39,90/mês
    </span>
    <p
      class="card__description"
    >
      Versão Start da Fast trade na medida para atender o trader que procura
   custo benefício aliados a boas funcionalidades gráficas, ferramentas
   como Times & Trades, Chart Trading, Resumo das Bolsas entre outros.
    </p>
    <span
      class="card__subtitle "
    >
      *Demonstrativo: 7 dias
    </span>
    <a
      class="link link__contained undefined"
      href="https://corretora.miraeasset.com.br/Home/Login"
    >
      Contratar
    </a>
  </li>
</div>
`;
